pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -148,22 +179,27 @@ public synchronized void readEntriesComplete(final List<Entry> entries, Object o
 
     @Override
     public synchronized void consumerFlow(Consumer consumer, int additionalNumberOfMessages) {
-        if (!havePendingRead) {
-            if (ACTIVE_CONSUMER_UPDATER.get(this) == consumer) {
-                if (log.isDebugEnabled()) {
-                    log.debug("[{}-{}] Trigger new read after receiving flow control message", name, consumer);
-                }
-                readMoreEntries(consumer);
-            } else {
-                if (log.isDebugEnabled()) {
-                    log.debug("[{}-{}] Ignoring flow control message since consumer is not active partition consumer",
-                            name, consumer);
-                }
+        if (havePendingRead) {
+            if (log.isDebugEnabled()) {
+                log.debug("[{}-{}] Ignoring flow control message since we already have a pending read req", name,
+                        consumer);
+            }
+        } else if (ACTIVE_CONSUMER_UPDATER.get(this) != consumer) {
+            if (log.isDebugEnabled()) {
+                log.debug("[{}-{}] Ignoring flow control message since consumer is not active partition consumer", name,
+                        consumer);
+            }
+        } else if (readOnActiveConsumerTask != null && (!readOnActiveConsumerTask.isDone()
+                || readOnActiveConsumerTask.isCancelled())) {
```

Comment:
if it's `readOnActiveConsumerTask.isCancelled()` then task will not be triggered -> then we are ignoring here and not reading entries? it's little confusing..
I think if we don't cancel task at line#96 then we don't need to check this condition.?

Reply:
I've removed cancellation process.